### PR TITLE
aws-staging-1: separate worker pool for free jobs on com (part 1)

### PR DESCRIPTION
**Pull request by igorwwwwwwwwwwwwwwwwwwww**
Part 1 of this change extracts cyclist out into its own module, but leaves the structure unchanged.

refs https://github.com/travis-ci/reliability/issues/116

### CODE REVIEW

**Comment by igorwwwwwwwwwwwwwwwwwwww** [comment]
This is trying to create two heroku apps called `cyclist-staging-1-com`, and I expect some other clashes down the line. We may need to introduce some extra suffix to disambiguate (I don't really want to set the "site" to `com-free`, that seems icky).

**Comment by meatballhat** [comment]
Higher level shared one SGTM, if possible :heart_eyes_cat:

**Comment by igorwwwwwwwwwwwwwwwwwwww** [comment]
The `make plan` suggests that this patch will cause the cyclist heroku apps to be re-created. The code is stateless, so that should be ok -- but it's also going to re-create the redis instance. Not sure if there is a good way of migrating that over.

**Comment by igorwwwwwwwwwwwwwwwwwwww** [comment]
I tried to use `terraform import` to keep the existing heroku apps, but wasn't able to get it to work. Possibly because the resources still exist currently and are being moved as opposed to added.

I managed to get it working via `terraform state mv` though:

```
➜  ~ terraform state mv module.aws_asg_com.heroku_addon.cyclist_redis module.aws_cyclist_com.heroku_addon.cyclist_redis
➜  ~ terraform state mv module.aws_asg_org.heroku_addon.cyclist_redis module.aws_cyclist_org.heroku_addon.cyclist_redis

➜  ~ terraform state mv module.aws_asg_com.heroku_app.cyclist module.aws_cyclist_com.heroku_app.cyclist
➜  ~ terraform state mv module.aws_asg_org.heroku_app.cyclist module.aws_cyclist_org.heroku_app.cyclist
```

`make plan` is now no longer trying to delete the heroku app or addon! 👍

**Comment by igorwwwwwwwwwwwwwwwwwwww** [comment]
Running into this issue in `aws-production-2`:

```
Error: Error running plan: 1 error(s) occurred:

* module.aws_asg_com.aws_launch_configuration.workers: expected length of user_data to be in the range (1 - 16384), got <snip>
```

The current user_data is 16389 bytes -- that's 5 bytes too many.

File: aws-production-2/main.tf [hunk 1]
```diff
@@ -273,10 +291,9 @@ module "aws_asg_org" {
 module "aws_asg_org_canary" {
   source                     = "../modules/aws_asg_canary"
   cyclist_auth_token         = "${random_id.cyclist_token_org.hex}"
-  cyclist_url                = "${module.aws_asg_org.cyclist_url}"
+  cyclist_url                = "${module.aws_cyclist_com.cyclist_url}"
```
Comment (meatballhat) [inline aws-production-2/main.tf]: Is this correct for `aws_asg_org_canary` to use the `aws_cyclist_com` cyclist?

File: modules/aws_asg/main.tf [hunk 2]
```diff
@@ -556,7 +447,7 @@ resource "aws_cloudwatch_dashboard" "main" {
         "metrics": [
           [
             "Travis/${var.site}${var.env == "staging" ? "-staging" : ""}",
-            "v1.travis.rabbitmq.consumers.builds.ec2.headroom",
+            "v1.travis.rabbitmq.consumers.builds.${var.worker_queue}.headroom",
```
Comment (meatballhat) [inline modules/aws_asg/main.tf]: :clap:

File: aws-production-2/main.tf (hunk 1 above)
Comment (igorwwwwwwwwwwwwwwwwwwww) [reply aws-production-2/main.tf]: good catch!

**Comment by igorwwwwwwwwwwwwwwwwwwww** [comment]
Some more `terraform state mv`s:

```
terraform state mv module.aws_asg_com.heroku_drain.cyclist_drain module.aws_cyclist_com.heroku_drain.cyclist_drain
terraform state mv module.aws_asg_org.heroku_drain.cyclist_drain module.aws_cyclist_org.heroku_drain.cyclist_drain
```

**Comment by igorwwwwwwwwwwwwwwwwwwww** [comment]
Tried to apply this on staging, it looks like there's still some naming conflicts that need to be disambiguated:

```
Error: Error applying plan:

2 error(s) occurred:

* module.aws_asg_com_free.aws_autoscaling_group.workers: 1 error(s) occurred:

* aws_autoscaling_group.workers: Error creating AutoScaling Group: AlreadyExists: AutoScalingGroup by this name already exists - A group with the name staging-1-workers-com already exists
	status code: 400, request id: 750095c0-746d-11e8-aed6-f36c1bb8eec2
* module.aws_asg_com_free.aws_iam_role.workers_sns: 1 error(s) occurred:

* aws_iam_role.workers_sns: Error creating IAM Role staging-1-workers-com-sns: EntityAlreadyExists: Role with name staging-1-workers-com-sns already exists.
	status code: 409, request id: 7281b3c7-746d-11e8-b164-8dfc36741a2c
```

**Comment by igorwwwwwwwwwwwwwwwwwwww** [comment]
I'm going to remove `aws_asg_com_free` for now, so that that change can be made more incrementally.

**Comment by igorwwwwwwwwwwwwwwwwwwww** [comment]
A couple more state moves required for production:

```
terraform state mv module.aws_asg_cs50.heroku_addon.cyclist_redis module.aws_cyclist_cs50.heroku_addon.cyclist_redis
terraform state mv module.aws_asg_cs50.heroku_app.cyclist module.aws_cyclist_cs50.heroku_app.cyclist
terraform state mv module.aws_asg_cs50.heroku_drain.cyclist_drain module.aws_cyclist_cs50.heroku_drain.cyclist_drain
terraform state mv module.aws_asg_cs50.aws_iam_user.cyclist module.aws_cyclist_cs50.aws_iam_user.cyclist
terraform state mv module.aws_asg_cs50.aws_iam_access_key.cyclist module.aws_cyclist_cs50.aws_iam_access_key.cyclist
terraform state mv module.aws_asg_cs50.aws_iam_user_policy.cyclist_actions module.aws_cyclist_cs50.aws_iam_user_policy.cyclist_actions
terraform state mv module.aws_asg_cs50.null_resource.cyclist module.aws_cyclist_cs50.null_resource.cyclist
```

**Comment by igorwwwwwwwwwwwwwwwwwwww** [comment]
Applied to staging and production!

**Comment by igorwwwwwwwwwwwwwwwwwwww** [comment]
Did see some of these though:

```
module.aws_cyclist_cs50.null_resource.cyclist (local-exec): -----> makey-go app detected
module.aws_cyclist_cs50.null_resource.cyclist (local-exec): -----> Downloading https://s3.amazonaws.com/travis-ci-cyclist-artifacts/travis-ci/cyclist/v0.5.0/build.tar.gz
module.aws_cyclist_org.null_resource.cyclist (local-exec): -----> makey-go app detected
module.aws_cyclist_com.null_resource.cyclist (local-exec): -----> makey-go app detected
module.aws_cyclist_org.null_resource.cyclist (local-exec): -----> Downloading https://s3.amazonaws.com/travis-ci-cyclist-artifacts/travis-ci/cyclist/v0.5.0/build.tar.gz
module.aws_cyclist_cs50.null_resource.cyclist (local-exec):        curl: (22) The requested URL returned error: 404 Not Found
module.aws_cyclist_cs50.null_resource.cyclist (local-exec):  !     Push rejected, failed to compile makey-go app.

module.aws_cyclist_cs50.null_resource.cyclist (local-exec):  !     Push failed
```

Looks like no artifact has been built for the v0.5.0 tag.

**Pull request merged**